tests/unit/llms/providers/test_azure_openai.py

```diff
@@ -27,10 +27,13 @@ def provider_config():
     )
 
 
+def setup_module(module):
```

Comment:
shouldn't this have a @pytest.fixture(scope="module") decorator?  or is "setup_module" a magic name?

also the previous behavior was invoking creating a new config for each test, now it's only going to happen for the module?

Reply:
Similar reason as above. The `setup_module` is magic, but it is executed only once per module. It works here, as these tests are not changing config values.
These tests actually doesn't depend on config at all, the config setting was here (as in many other tests) because we didn't had defaults for subconfigs before.

~~Again, to be on safe side, I changed it to function-scoped fixture to ensure the config is initialized fresh for each test.~~

I added `conftest.py` into unit and integration package (we could have just one contest in the tests root, but we are running tests per package) with fixture that is executed **before every test** and sets the config as empty.
With this, we can ensure that if some test make changes in the config, it stay "isolated" as each test gets a fresh new one by default.

So I cleaned test a bit from the extra config setting that is not needed with this change. And there was just one test that was failing as was depending on the config state from the other test - so we are good now.